[DOCS] DOCS-369 sql data connector configuration guide

Rachel-Reverie opened the pull request:
Changes proposed in this pull request:
- Adds a guide for configuring a SQL based datasource, using either an inferred, configured, or runtime data connector.

### Definition of Done
- [x] I have made corresponding changes to the documentation

Thank you for submitting!

Rachel-Reverie committed:
DOC-369
Draft up to section 7's ConfiguredAssetDataConnector tab.

Rachel-Reverie committed:
DOC-369
Draft up to section 7's Inferred and Runtime tabs.

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=6002&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=6002&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/6002/fd4ff3d8/09b5be94d768a7474d65c98fe374007f88bffa4d.svg)](https://app.codesee.io/r/reviews?pr=6002&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

Rachel-Reverie committed:
DOC-369
Rollback inadvertant changes to the notebook.

Rachel-Reverie committed:
DOC-369
Rollback inadvertent changes to the notebook., take 2

Rachel-Reverie committed:
DOC-369
Rollback inadvertent changes to the notebook., take 3

Rachel-Reverie committed:
DOC-369
Manual replacement of notebook with contents from develop

Rachel-Reverie committed:
DOC-369
Again with the revert

Rachel-Reverie committed:
DOC-369
And once again with the manual replacement of the content.

Rachel-Reverie committed:
Merge branch 'develop' into d/docs-369/sql_data_connector_configuration_guide

Rachel-Reverie committed:
DOC-369
Draft for the remaining portions of the SQL data connector guide

Rachel-Reverie committed:
Merge remote-tracking branch 'origin/d/docs-369/sql_data_connector_configuration_guide' into d/docs-369/sql_data_connector_configuration_guide

Rachel-Reverie committed:
DOC-369
Fix typo in relative link to credentials guide

Rachel-Reverie committed:
DOC-369
- clarification of tables in Additional Notes
- Changed SimpleSqlAlchemy in heading to SqlAlchemy
- Removed `table_name` parameter from splitter_method table
- Indicated splitting and sampling portion of guide in ToC

Rachel-Reverie committed:
DOC-371
- added splitting examples into Step 8.

AFineDayFor commented on docs/guides/connecting_to_your_data/datasource_configuration/components/_section_intro.mdx:
```diff
@@ -1,4 +1,4 @@
 <!--- props.backend: The name of the backend used by the Datasource's Execution Engine. -->
 <p>
-This guide will walk you through the process of configuring a {props.backend} Datasource from scratch, verifying that your configuration is valid, and adding it to your Data Context.  By the end of this guide you will have a {props.backend} Datasource which you can use in future workflows for creating Expectations and validating data.
+This guide will walk you through the process of configuring a {props.backend} Datasource from scratch, verifying that your configuration is valid, and adding it to your Data Context.  By the end of this guide you will have a {props.backend} Datasource which you can use in future workflows for creating Expectations and Validating data.
```
NIT: The use of validating as a proper noun in contrast to Validator as a class or proper noun is something that stands out, curious about your thoughts on this too.

Rachel-Reverie replied:
It's capitalized because we have a technical term entry for "The act of applying an Expectation Suite to a Batch" (the entry is technically for the word "Validation", but I've been applying it to all variations of the Validate verb as well) and we always capitalize our technical terms.  That will be more evident when I tag all of our technical terms in a future PR.

AFineDayFor commented on docs/guides/connecting_to_your_data/datasource_configuration/how_to_configure_a_sql_datasource.md:
```diff
@@ -0,0 +1,98 @@
+
+<SectionNextSteps />
+
+## Additional notes
```
nit: beautiful templating i think :) else your works just pristine by nature; either way this isn't used here.

talagluck commented on docs/guides/connecting_to_your_data/datasource_configuration/sql_components/_part_data_connector_required_keys_overview.mdx:
```diff
@@ -0,0 +1,29 @@
+
+<div>
+    <p>
+        When defining an <code>{props.data_connector_type}</code> you will need to provide values for { props.inferred && <span>one required key</span> || <span>two required keys</span> } in the Data Connector's configuration dictionary (the currently empty dictionary that corresponds to <code>"{props.data_connector_name}"</code> in the example above).  { props.inferred && <span>This key/value pa ...
```
```suggestion
        When defining a(n) <code>{props.data_connector_type}</code> you will need to provide values for { props.inferred && <span>one required key</span> || <span>two required keys</span> } in the Data Connector's configuration dictionary (the currently empty dictionary that corresponds to <code>"{props.data_connector_name}"</code> in the example above).  { props.inferred && <span>This key/value pair consists of:</span> || <span>These key/value pairs consist of:</span> }
```

Rachel-Reverie replied:
I implemented a different correction so that a/an would change appropriately, instead of having them combined as a(n).

talagluck commented on docs/guides/connecting_to_your_data/datasource_configuration/sql_components/_table_splitting_methods.mdx:
```diff
@@ -0,0 +1,23 @@
+
+#### Available methods for `splitting_method` values and their configuration parameters:
+
+| `splitting_method`                | `splitting_kwargs`                                                       | Returned Batch Data                                                                                                                                                                                           |
```
```suggestion
| `splitter_method`                | `splitter_kwargs`                                                       | Returned Batch Data                                                                                                                                                                                           |
```

Rachel-Reverie replied:
I corrected a few more instances in that section where I had used `*ing_*` instead of `*er_*`.

talagluck reviewed (commented):
Looks great overall! A couple of minor changes, one of which should be addressed before merge.

Rachel-Reverie committed:
DOC-371
- correct "splitting_method" and "splitting_kwargs" to "splitter_*"

Rachel-Reverie committed:
DOC-371
- correct usage of a vs an in dynamic paragraph.

AFineDayFor reviewed (approved):
Rev 2/2. Only remarks are a nit of a capital V in Validating at this time.

Rachel-Reverie committed:
Merge branch 'develop' into d/docs-369/sql_data_connector_configuration_guide
